packages/protocol/contracts/layer1/preconf/impl/PreconfRouter.sol

```diff
@@ -4,33 +4,95 @@ pragma solidity ^0.8.24;
 import "../iface/IPreconfRouter.sol";
 import "../iface/IPreconfWhitelist.sol";
 import "src/layer1/based/ITaikoInbox.sol";
+import "src/shared/libs/LibAddress.sol";
 import "src/shared/libs/LibStrings.sol";
 import "src/shared/common/EssentialContract.sol";
 
 /// @title PreconfRouter
 /// @custom:security-contact security@taiko.xyz
 contract PreconfRouter is EssentialContract, IPreconfRouter {
-    uint256[50] private __gap;
+    mapping(bytes32 => ForcedTx) public forcedTxLists;
 
-    constructor(address _resolver) EssentialContract(_resolver) { }
+    uint256 public pendingForcedTxHashes;
+    
+    uint256 public inclusionWindow;
+
+    uint256 public baseStakeAmount;
+
+    uint256[46] private __gap;
+
+    constructor(address _resolver, uint256 _inclusionWindow, uint256 _baseStakeAmount) EssentialContract(_resolver) {
+        inclusionWindow = _inclusionWindow;
+        baseStakeAmount = _baseStakeAmount;
+    }
 
     function init(address _owner) external initializer {
         __Essential_init(_owner);
     }
 
+    function updateBaseStakeAmount(uint256 _newBaseStakeAmount) external onlyOwner {
+        baseStakeAmount = _newBaseStakeAmount;
+    }
+
+    function getRequiredStakeAmount() public view returns (uint256) {
+        uint256 multiplier = pendingForcedTxHashes == 0 ? 1 : pendingForcedTxHashes;
+        return baseStakeAmount * multiplier;
+    }
+
+    function storeForcedTx(bytes calldata _txList) payable external {
+        uint256 requiredStake = getRequiredStakeAmount();
+        require(msg.value >= requiredStake, InsufficientStakeAmount());
```

Comment:
```suggestion
        require(msg.value == requiredStake, InsufficientStakeAmount());
```

Reply:
Otherwise, please send back the overpaid fee.

Reply:
will add